### DISCUSSION TITLE
unit and num processors

### DISCUSSION THREAD

**Discussion by Salma-Mao** [idaholab/moose#20464] [Q&A General] [answered]
Hello everyone!
How  can I change the units of mesh in my model?  Such as, unit of mesh in the project [Case2DegradEigenBFI3_4um.i](https://github.com/ngrilli/c_pfor_am/blob/main/test/tests/Paper_ElementEliminationMethod/Case2DegradEigenBFI3_4um.i) is micron.  
When I compile the file, I find that num processors is always 1. How can I change this to improve my computing speed.

Parallelism:
  Num Processors:          1
  Num Threads:             1

Thanks

**Comment by GiudGiud** [comment, marked as answer]
Hello

if the mesh is in micron, then you will have to make sure every material property and so on is in micron. There are utilities to help you do this from the input file. I ll post a link later. 

Which module are you trying to use? Some have much better support for different unit systems than others. 

To change the number of processors you have to use MPI. 
run `mpirun -n XX app-opt -i input_file.i`
where `app` is you app, `XX` is the number of processors you want to use. 

Guillaume

**Reply by GiudGiud** [reply]
here are two pages about units in moose
https://mooseframework.inl.gov/source/utils/Units.html
and in the input file:
https://mooseframework.inl.gov/application_usage/input_syntax.html

**Reply by Salma-Mao** [reply]
Thanks for  your help.
